server/src/main/java/io/crate/metadata/settings/NumberOfReplicas.java

```diff
@@ -19,51 +19,76 @@
+ * <ul>
+ *  <li>index.number_of_replicas - an integer</li>
+ *  <li>index.auto_expand_replicas - "false" or a range like "0-1" or "0-all" </li>
+ * </ul>
+ **/
+public class NumberOfReplicas extends Setting<Settings> {
 
     private static final Pattern EXPAND_REPLICA_PATTERN = Pattern.compile("\\d+\\-(all|\\d+)");
-    private final String esSettingKey;
-    private final String esSettingsValue;
+    private static final Settings DEFAULT = Settings.builder()
+        .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 0)
+        .put(AutoExpandReplicas.SETTING_KEY, AutoExpandReplicas.SETTING.getDefaultRaw(Settings.EMPTY))
+        .build();
 
-    public NumberOfReplicas(Integer numReplicas) {
-        this.esSettingKey = IndexMetadata.SETTING_NUMBER_OF_REPLICAS;
-        this.esSettingsValue = numReplicas.toString();
+    public static final NumberOfReplicas SETTING = new NumberOfReplicas();
+
+    private NumberOfReplicas() {
+        super(
+            new SimpleKey(IndexMetadata.SETTING_NUMBER_OF_REPLICAS),
+            (s) -> "0-1",
+            NumberOfReplicas::parseValue,
+            DataTypes.STRING,
+            Property.ReplicatedIndexScope
+        );
     }
 
-    public NumberOfReplicas(String numReplicas) {
-        assert numReplicas != null : "numReplicas must not be null";
-        validateExpandReplicaSetting(numReplicas);
+    /**
+     * Converts the external value (`0` or `0-1`) to the internal index. settings.
+     */
+    private static Settings parseValue(String value) {
+        Settings.Builder builder = Settings.builder();
 
-        this.esSettingKey = AutoExpandReplicas.SETTING_KEY;
-        this.esSettingsValue = numReplicas;
-    }
+        // in case the number_of_replicas is changing from auto_expand to a fixed number -> disable auto expand
+        builder.put(AutoExpandReplicas.SETTING.getKey(), false);
 
-    private static void validateExpandReplicaSetting(String replicas) {
-        if (!EXPAND_REPLICA_PATTERN.matcher(replicas).matches()) {
-            throw new IllegalArgumentException("The \"number_of_replicas\" range \"" + replicas + "\" isn't valid");
+        try {
+            int numReplicas = parseInt(value, 0, IndexMetadata.SETTING_NUMBER_OF_REPLICAS);
+            builder.put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, numReplicas);
+        } catch (NumberFormatException e) {
+            validateExpandReplicaSetting(value);
+            builder.put(AutoExpandReplicas.SETTING_KEY, value);
         }
-    }
 
-    public String esSettingKey() {
-        return esSettingKey;
+        return builder.build();
     }
 
-    public String esSettingValue() {
-        return esSettingsValue;
+    @Override
+    public Settings getDefault(Settings settings) {
+        return DEFAULT;
     }
 
-    public static String fromSettings(Settings settings) {
+    /**
+     * Reverse of {@link #parseValue(String). Converts index. settings into the public value
+     **/
```

Comment:
Maybe obvious but for extra clarity could add that it's either a value or a range